KOGITO-6868 Added documentation for Source Files Addon (#4199)
kiegroup/kie-docs: doc-content/kogito-docs/src/main/asciidoc/creating-running/chap-kogito-creating-running.adoc

</dependency>
----

[#source-files-add-on]
=== {PRODUCT} source files add-on

> The {PRODUCT} Source Files Add-on adds the capability of listing and downloading source files to {PRODUCT} projects.

Reviewer: ```suggestion
The {PRODUCT} source files add-on adds the capability of listing and downloading source files to {PRODUCT} projects.
```

> > **⚠️**

Reviewer: Is this a note or important notice? Instead of `**⚠️**` you can write in the `[Important]` syntax.

> This add-on isn't supposed to be used by end-users, if not needed for internal integrations, since it can be a potential security breach for some users' use cases.

Reviewer: ```suggestion
[Important]
====
This add-on is not supposed to be used by end-users, if not needed for internal integrations, since it can be a potential security breach for some users' use cases.
====
```

</dependency>
----

Access to all the endpoints is granted to the role `source-files-client`.
To define basic authentication, you can add the following code to your `application.properties` file:

.Example of credentials for the user `scott` and password `jb0ss`
[source,properties]
----
quarkus.security.users.embedded.enabled=true
quarkus.security.users.embedded.plain-text=true
quarkus.security.users.embedded.users.scott=jb0ss
quarkus.security.users.embedded.roles.scott=source-files-client
----

You can also disable plain-text passwords by setting the `quarkus.security.users.embedded.plain-text` property to
`false` or just removing it from the configuration. In this case, the passwords are encrypted using MD5.

.Example of credentials for the user `scott` and password `jb0ss` encrypted using MD5
[source,properties]
----
quarkus.security.users.embedded.enabled=true
quarkus.security.users.embedded.users.scott=4960d03c13b5aff4ca6886758a8be019
quarkus.security.users.embedded.roles.scott=source-files-client
----

To define a more robust authentication mechanism, you can refer to the https://quarkus.io/guides/security[Quarkus documentation].

> The Source Files Add-on provides REST support for the following operation:

Reviewer: ```suggestion
The source files add-on provides REST support for the following operation:
```
